src/ProgressOnderwijsUtils.Analyzers/MaybeStateAnalyzer.cs

```diff
@@ -1,43 +0,0 @@
-﻿using System.Collections.Immutable;
-using Microsoft.CodeAnalysis;
-using Microsoft.CodeAnalysis.CSharp;
-using Microsoft.CodeAnalysis.Diagnostics;
-
-namespace ProgressOnderwijsUtils.Analyzers
-{
-    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
-    public sealed class MaybeStateAnalyzer : DiagnosticAnalyzer
-    {
-        public static readonly DiagnosticDescriptor Rule = new DiagnosticDescriptor(
-            "POU2001",
-            "Maybe state is not allowed to be another Maybe",
-            "The maybe-error state must not be ignored.",
-            "Functional",
-            DiagnosticSeverity.Error,
-            true
-        );
-
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
-            => ImmutableArray.Create(Rule);
-
-        public override void Initialize(AnalysisContext context)
-        {
-            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze);
-            context.EnableConcurrentExecution();
-            context.RegisterSyntaxNodeAction(
-                Analyze,
-                SyntaxKind.InvocationExpression

```

Comment:
Wat vast ook had gewerkt (maar zo prima!) was hier *voor* de semantic model aan een check op de (syntactische) method name, zodat je de prijs van de semantic model niet voor elke method-call hoeft te betalen.

Reply:
Had ik wel aan gedacht, maar dan gaan we dus alleen die WhenOk controleren, en dan kunnen we net zo goed die obsolete overload maken.